File: src/javascript/user/message/get-message-image.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const messageID = process.env.FAXCORE_MESSAGE_ID;
if (!messageID) {
  throw new Error('FAXCORE_MESSAGE_ID is required.');
}

const client = new FaxCoreClient();
const result = await client.get('/api/message/image', {
  'model.messageID': messageID,
  'model.xsactSeq': process.env.FAXCORE_XSACT_SEQ || 1,
  'model.page': process.env.FAXCORE_IMAGE_PAGE || 1,
  'model.width': process.env.FAXCORE_IMAGE_WIDTH || 800
});

printResult('Message image', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/GetMessageImage.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class GetMessageImage
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            if (string.IsNullOrWhiteSpace(config.MessageId))
            {
                throw new InvalidOperationException("Set FAXCORE_MESSAGE_ID before running user.message.image.");
            }

            var path = string.Format(
                "/api/message/image?model.messageID={0}&model.xsactSeq={1}&model.page={2}&model.width={3}",
                Uri.EscapeDataString(config.MessageId),
                config.XsactSeq,
                config.ImagePage,
                config.ImageWidth);

            return client.GetAsync(path);
        }
    }
}

